### PR TITLE
7054 Fetch report lineage by report ID

### PR DESCRIPTION
**Pull request by carlosfelix2**
This PR makes it so the first processing step in a pipeline is not required to generate a report lineage, so the submission history can return destinations.

How the previous query worked: The previous query relied on an action having a report lineage. If you submit a new COVID report and use the action ID on the report lineage table you will find that it has an entry for that action ID since the COVID pipeline generates an internal CSV upon receipt of new data, hence adding a new entry in the report lineage table.  This made it easy to find the lineage based on the submission action ID.

How it works now: The UP does not generate any new reports in its first convert step, so there will be an entry in report file denoting the submitted report, but none in the report lineage table.  The report lineage table will contain lineage starting with the next action in the pipeline, which means the old query could not find the lineage and hence could not generate destinations for the response.  We use here the report ID instead to find the initial lineage for a given report.

Test Steps:
1. Build and run RS for this branch.
2. Submit both a COVID and full-ELR report to your local RS.
3. Get the submission details for both reports and verify destinations are included in the response.

## Changes
- Updated the query to get the report lineage

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/main/kotlin/history/azure/DatabaseSubmissionsAccess.kt [hunk 1]
```diff
@@ -124,17 +128,33 @@ class DatabaseSubmissionsAccess(
         )
     }
 
-    override fun <T> fetchRelatedActions(actionId: Long, klass: Class<T>): List<T> {
-        val cte = reportDescendantExpression(actionId)
+    override fun <T> fetchRelatedActions(reportId: UUID, klass: Class<T>): List<T> {
```
Comment (clediggins-usds) [inline prime-router/src/main/kotlin/history/azure/DatabaseSubmissionsAccess.kt:131]: Please add doc to explain what this is doing.

File: prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt [hunk 2]
```diff
@@ -91,15 +92,7 @@ class DatabaseDeliveryAccess(
...
-    /**
-     * Fetch the details of an action's relations (descendants).
-     * This is done through a recursive query on the report_lineage table.
-     *
-     * @param actionId the action id attached to the action to find relations for.
-     * @param klass the class that the found data will be converted to.
-     * @return a list of descendants for the given action id.
-     */
-    override fun <T> fetchRelatedActions(actionId: Long, klass: Class<T>): List<T> {
+    override fun <T> fetchRelatedActions(reportId: UUID, klass: Class<T>): List<T> {
```
Comment (clediggins-usds) [inline prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt:95]: kdoc

Comment (carlosfelix2) [reply prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt:95]: The kdoc gets inherited from the base class.

File: prime-router/src/main/kotlin/history/azure/DatabaseSubmissionsAccess.kt (hunk 1 above)
Comment (carlosfelix2) [reply prime-router/src/main/kotlin/history/azure/DatabaseSubmissionsAccess.kt:131]: The kdoc gets inherited from the base class

**Review by clediggins-usds — APPROVED** [review]

File: prime-router/src/main/kotlin/history/azure/DatabaseSubmissionsAccess.kt [hunk 3]
```diff
@@ -124,17 +128,33 @@ class DatabaseSubmissionsAccess(
...
+            // action ID.
+            val actionId = try {
+                DSL.using(txn)
+                    .selectDistinct(REPORT_LINEAGE.ACTION_ID)
+                    .from(REPORT_LINEAGE)
+                    .where(REPORT_LINEAGE.PARENT_REPORT_ID.eq(reportId))
+                    .fetchOneInto(Long::class.java)
+            } catch (e: TooManyRowsException) {
+                logger.warn("Invalid report file lineage with multiple action IDs for parent report $reportId")
+                null
```
Comment (TurkeyFried) [inline prime-router/src/main/kotlin/history/azure/DatabaseSubmissionsAccess.kt:145]: is it ok that we return null here vs returning emptyList below?

**Pull request merged**